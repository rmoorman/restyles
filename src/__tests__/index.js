/* eslint-env jest */
/* eslint-disable jsx-quotes */
import React from 'react'
import renderer from 'react-test-renderer'
import {matcher, serializer} from 'jest-glamor-react'
import Style from '../index'

expect.addSnapshotSerializer(serializer)
expect.extend(matcher)

describe('restyles', () => {
  test('renders elements with styles', () => {
    const props = {
      online: true,
      theme: {radius: 5}
    }

    const tree = renderer
      .create(
        <Style
          css={{
            color: 'blue',
            ':hover': {
              color: 'red'
            },
            '.profile': {
              color: props.online ? 'green' : 'gray',
              fontSize: 20
            }
          }}
        >
          {cls => (
            <div className={cls}>
              This will be blue until hovered.
              <div className="profile">
                This font size will be 20px
              </div>
            </div>
          )}
        </Style>
      )
      .toJSON()

    expect(tree).toMatchSnapshotWithGlamor()
  })

  test('renders crazy stuff', () => {
    const tree = renderer
      .create(
        <Style
          css={{
            color: 'blue',
            ' .profile': {
              color: 'red',
              ' .username': {
                color: 'green',
                ' .three': {
                  color: 'gray',
                  ' .inner': {
                    color: 'rebeccapurple',
                    fontSize: 20
                  }
                }
              }
            },
            ':hover': {
              ' .one': {
                color: 'blue',
                ' .two': {
                  color: 'gray',
                  ' .three': {
                    color: 'black',
                    ' .inner': {
                      color: 'purple',
                      fontSize: 20
                    }
                  }
                }
              }
            },
            ':active': {
              ' .one': {
                color: 'blue',
                ' .two': {
                  color: 'gray',
                  ' .three': {
                    color: 'black',
                    ' .inner': {
                      color: 'purple',
                      fontSize: 20
                    }
                  }
                }
              }
            },
            ':before': {
              ' .one': {
                color: 'blue',
                ' .two': {
                  color: 'gray',
                  ' .three': {
                    color: 'black',
                    ' .inner': {
                      color: 'purple',
                      fontSize: 20
                    }
                  }
                }
              }
            },
            '@media (min-width: 500px) and (orientation: landscape)': {
              ' .one': {
                color: 'blue',
                ' .two': {
                  color: 'gray',
                  ' .three': {
                    color: 'black',
                    ' .inner': {
                      color: 'purple',
                      fontSize: 20
                    }
                  }
                }
              }
            }
          }}
        >
          {cls => (
            <div className={cls}>
              blue
              <div className={'profile'}>
                red
                <div className={'username'}>
                  green
                  <div className={'three'}>
                    <span className={'inner'}>test</span>
                  </div>
                </div>
              </div>
            </div>
          )}
        </Style>
      )
      .toJSON()

    expect(tree).toMatchSnapshotWithGlamor()
  })
})
