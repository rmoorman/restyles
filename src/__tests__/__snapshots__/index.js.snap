// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`restyles renders crazy stuff 1`] = `
.c-blue {
  color: blue;
}

.__pprofile-c-red  .profile {
  color: red;
}

._1yp8db4  .profile  .username {
  color: green;
}

._pczu08  .profile  .username  .three {
  color: gray;
}

._nfl9mc  .profile  .username  .three  .inner {
  color: rebeccapurple;
}

._1q1tm5b  .profile  .username  .three  .inner {
  font-size: 20px;
}

._hover__pone-c-blue:hover  .one {
  color: blue;
}

._1plqem2:hover  .one  .two {
  color: gray;
}

._1idc1ym:hover  .one  .two  .three {
  color: black;
}

._bp7izt:hover  .one  .two  .three  .inner {
  color: purple;
}

._sahbn7:hover  .one  .two  .three  .inner {
  font-size: 20px;
}

._active__pone-c-blue:active  .one {
  color: blue;
}

._5oxw7k:active  .one  .two {
  color: gray;
}

._1jduipw:active  .one  .two  .three {
  color: black;
}

._14tsqoz:active  .one  .two  .three  .inner {
  color: purple;
}

._8n05kp:active  .one  .two  .three  .inner {
  font-size: 20px;
}

._before__pone-c-blue:before  .one {
  color: blue;
}

._86dbmt:before  .one  .two {
  color: gray;
}

._1gtudkh:before  .one  .two  .three {
  color: black;
}

._177n1w6:before  .one  .two  .three  .inner {
  color: purple;
}

._b9qsss:before  .one  .two  .three  .inner {
  font-size: 20px;
}

@media (min-width: 500px) and (orientation: landscape) {
  ._4b7p1m  .one {
    color: blue;
  }
}

@media (min-width: 500px) and (orientation: landscape) {
  ._eulomd  .one  .two {
    color: gray;
  }
}

@media (min-width: 500px) and (orientation: landscape) {
  ._2w2ir5  .one  .two  .three {
    color: black;
  }
}

@media (min-width: 500px) and (orientation: landscape) {
  ._13s6nna  .one  .two  .three  .inner {
    color: purple;
  }
}

@media (min-width: 500px) and (orientation: landscape) {
  ._ixb23w  .one  .two  .three  .inner {
    font-size: 20px;
  }
}

<div
  className="c-blue __pprofile-c-red _1yp8db4 _pczu08 _nfl9mc _1q1tm5b _hover__pone-c-blue _1plqem2 _1idc1ym _bp7izt _sahbn7 _active__pone-c-blue _5oxw7k _1jduipw _14tsqoz _8n05kp _before__pone-c-blue _86dbmt _1gtudkh _177n1w6 _b9qsss _4b7p1m _eulomd _2w2ir5 _13s6nna _ixb23w"
>
  blue
  <div
    className="profile"
  >
    red
    <div
      className="username"
    >
      green
      <div
        className="three"
      >
        <span
          className="inner"
        >
          test
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`restyles renders elements with styles 1`] = `
.c-blue {
  color: blue;
}

._hover-c-red:hover {
  color: red;
}

.__pprofile-c-green  .profile {
  color: green;
}

.__pprofile-fs-20  .profile {
  font-size: 20px;
}

<div
  className="c-blue _hover-c-red __pprofile-c-green __pprofile-fs-20"
>
  This will be blue until hovered.
  <div
    className="profile"
  >
    This font size will be 20px
  </div>
</div>
`;
